cypress/unit/store-mutations-setDataDictionary.cy.js

```diff
@@ -0,0 +1,75 @@
+import { mutations } from "~/store/index-refactor";
+
+let state = {};
+
+describe("setDataDictionary", () => {
+
+    beforeEach(() => {
+
+        // Setup
+        state = {
+
+            dataDictionary: {
+
+                "age": {
+
+                    "Description": "age of the participant",
+                    "Units": "years"
+                },
+                "sex": {
+
+                    "Description": "sex of the participant as reported by the participant",
+                    "Levels": {
+                        "M": "male",
+                        "F": "female"
+                    }
+                }
+            },
+
+            dataTable: [
+
+                {
+                    "age": "value1", "sex": "value2"
+                },
+                {
+                    "age": "value3", "sex": "value4"
+                }
+            ]
+        };
+    });
+
+    it("New columns in the uploaded dictionary should not be added to the state data dictionary?", () => {
+
+        // Act
+        mutations.setDataDictionary(state,
+            { "group": { "key1": "value1" } },
+            ["age", "sex"]);
+
+        // Assert
+        expect(state.dataDictionary).to.not.contain.keys("group");
+    });
+
+    it("New string/string key/value pairs for columns should replace values in the state data dictionary", () => {
+
+        // Act
+        mutations.setDataDictionary(state,
+            { "age": { "Units": "minutes" } },
+            ["age", "sex"]);
+
+        // Assert
+        expect(state.dataDictionary["age"]["Units"]).to.equal("minutes");
+    });
+
+    it("New string/object key/value pairs in the uploaded dictionary should update/retain values in the state data dictionary", () => {
+
+        // Act
+        mutations.setDataDictionary(state,
+            { "sex": { "Levels": { "M": "male", "F": "female", "NB": "non-binary" }} },
```

Comment:
Shouldn't we test values that are different from the mocked `dataDictionary` here? 
i.e. `"Levels": { "M": "a male participant" }` or something like that

Reply:
That would cover a case of 'update' as opposed to just a new key, yes. I'll make the change for `'M'`.